### DISCUSSION TITLE
thanks for typebox which enables end-to-end typesafety across an entire ecosystem

### DISCUSSION THREAD

**Discussion by samuelstroschein** [sinclairzx81/typebox#617] [Show and tell]
Hey @sinclairzx81, 

I just wanted to say thank you big time for typebox! 

It enables us to have end-to-end type safety with TS types, runtime validation, plus (key benefit!) "storage" validation via JSON schema. The storage validation possibility is essential because we can suddenly lint translation files [see this video](https://www.loom.com/share/a2b7f9e6d8954beca8dcb0a0a778b27a?sid=4ad2320c-862a-4dff-8720-da947c28a18c), settings files and much more as part of the [inlang ecosystem](https://inlang.com).

<img width="981" alt="CleanShot 2023-10-04 at 13 09 59@2x" src="https://github.com/sinclairzx81/typebox/assets/35429197/54d2b43c-ee64-4e18-b2c3-346e362e7032">

[Discord chat link](https://discord.com/channels/897438559458430986/1159079779685904385/1159084939967475743)

cc @openscript

**Comment by sinclairzx81** [comment]
@samuelstroschein Hi! Thank you for the kind words! :) It's great to hear people are getting good use from TB!

**Comment by aquisio** [comment]
I totally agree. I just discovered Typebox and it has transformed my code into something that is now a pleasure to work with.

Thanks for creating Typebox and for making it so robust and complete.

**Comment by oscar6echo** [comment]
Fully agree: I discovered typebox and its companion package typebox-codegen this year and they are SO useful !   :+1: 
I like the vision and the implementation - as much as I can judge.  

@sinclairzx81 congrats and thx
